### PR TITLE
Find policies/rules which apply to a certain workload

### PR DESCRIPTION
**Pull request by vihaanko**
Take a look at my code for Issue #12. I am yet to do the range check for the IP. So just take a look at what I have so far.

### CODE REVIEW

File: pyclient/apps/find_rule.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+                # Iterate over the "to" ip addresses of each rule
+                for to in rule.to_ip_addresses:
+                    if (inRange(interface_ip, to)):
+                        rule_match = True
+                        if (policy_match == False):
+                            policiesInRange.append("Policy: " + str(policy.meta.name))
+                            policy_match = True
+                if (rule_match == True):
+                    print(rule)
```
Comment (jainvipin) [inline pyclient/apps/find_rule.py:83]: seems like a debug print that sneaked in the output; either remove or put it under `--debug`

**Review by jainvipin — CHANGES_REQUESTED** [review]
@vihaanko - pls address some comments; thanks for the PR !!

File: pyclient/apps/find_rule.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
+import os
```
[the remark this answers is not in the document]
Comment (vihaanko) [reply pyclient/apps/find_rule.py:1]: Fixed

File: pyclient/apps/find_rule.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+client = api_client.ApiClient(configuration)
+security_instance = SecurityV1Api(client)
+workload_instance = WorkloadV1Api(client)
+
+
+# Set default variable
+setDefault = "default"
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--name', type=str, required=True)
```
[the remark this answers is not in the document]
Comment (vihaanko) [reply pyclient/apps/find_rule.py:43]: Fixed

File: pyclient/apps/find_rule.py [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+args = parser.parse_args()
+
+workload = workload_instance.get_workload(setDefault, args.name)
+# print(workload)
```
[the remark this answers is not in the document]
Comment (vihaanko) [reply pyclient/apps/find_rule.py:47]: Fixed

File: pyclient/apps/find_rule.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+policy_list = security_instance.list_network_security_policy(setDefault)
```
[the remark this answers is not in the document]
Comment (vihaanko) [reply pyclient/apps/find_rule.py:49]: Fixed

File: pyclient/apps/find_rule.py (hunk 1 above)
Comment (vihaanko) [reply pyclient/apps/find_rule.py:83]: Fixed

File: pyclient/apps/find_rule.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        # Iterate over policies
+        for policy in policy_list.items:
+            policy_match = False
+            rule_num = 0
+            # Iterate over rules in policy
+            for rule in policy.spec.rules:
+                rule_num += 1
+                rule_match = False
+
+                # Iterate over the "from" ip addresses of each rule
```
[the remark this answers is not in the document]
Comment (vihaanko) [reply pyclient/apps/find_rule.py:67]: Fixed

File: pyclient/apps/find_rule.py [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+                if (rule_match == True):
+                    print(rule)
+                    policiesInRange.append("Rule#" + str(rule_num) + ": " + printRule(rule))
+
+
+if (len(policiesInRange) > 0):
+    for i in policiesInRange:
+        print(i)
+else:
+    print("No policies include your ip address.")
```
[the remark this answers is not in the document]
Comment (vihaanko) [reply pyclient/apps/find_rule.py:91]: Fixed, a IP address can now be taken as an input.

**Review by jainvipin — APPROVED** [review]
LGTM

**Pull request merged**